xapian-letor/feature/tfdoclenfeature.cc

```diff
@@ -41,9 +40,9 @@ TfDoclenFeature::get_values() const
 {
     LOGCALL(API, std::vector<double>, "TfDoclenFeature::get_values", NO_ARGS);
 
-    Query query = Feature::internal->feature_query;
-    map<string, long int> tf = Feature::internal->termfreq();
-    map<string, long int> doc_len = Feature::internal->doc_length();
+    Query query = Feature::feature_query;
+    map<string, long int> tf = Feature::termfreq;
```

Comment:
I cannot eliminate the copying from FeatureList::Internal to Feature as it saves a lot of time which would be otherwise spent in recomputing the statistics in Feature. I can avoid the copy here but since its a const function, either I cannot avoid copying or I remove the const. Which would be a better option here ??. In my opinion having Feature::get_value() as a const function makes no sense.